src/stan/services/pathfinder/multi.hpp

```diff
@@ -0,0 +1,200 @@
+        }
+      });
+  const auto end_pathfinders_time = std::chrono::steady_clock::now();
+  const double pathfinders_delta_time
+      = std::chrono::duration_cast<std::chrono::milliseconds>(
+            end_pathfinders_time - start_pathfinders_time)
+            .count()
+        / 1000.0;
+  const auto start_psis_time = std::chrono::steady_clock::now();
+  // Because of failure in lp calcs we can have multiple returned sizes
+  size_t num_returned_samples = 0;
+  const size_t successful_pathfinders = individual_samples.size();
+  if (successful_pathfinders == 0) {
+    logger.info("No pathfinders ran successfully");
+    return error_codes::SOFTWARE;
+  }
+  if (refresh != 0) {
+    logger.info("Total Evaluations: (" + std::to_string(lp_calls) + ")");
+  }
+  for (size_t i = 0; i < successful_pathfinders; i++) {
+    num_returned_samples += individual_lp_ratios[i].size();
+  }
+  Eigen::Array<double, -1, 1> lp_ratios(num_returned_samples);
+  Eigen::Array<double, -1, -1> samples(individual_samples[0].rows(),
+                                       num_returned_samples);
+  Eigen::Index filling_start_row = 0;
+  for (size_t iter = 0; iter < successful_pathfinders; ++iter) {
+    const Eigen::Index individ_num_samples = individual_lp_ratios[iter].size();
+    lp_ratios.segment(filling_start_row, individ_num_samples)
+        = individual_lp_ratios[iter];
+    samples.middleCols(filling_start_row, individ_num_samples)
+        = individual_samples[iter];
+    filling_start_row += individ_num_samples;
+  }
+  const auto tail_len = std::min(0.2 * num_returned_samples,
+                                 3 * std::sqrt(num_returned_samples));
+  Eigen::Array<double, -1, 1> weight_vals
+      = stan::services::psis::psis_weights(lp_ratios, tail_len, logger);
+  boost::ecuyer1988 rng
+      = util::create_rng<boost::ecuyer1988>(random_seed, path);
+  boost::variate_generator<
+      boost::ecuyer1988&,
+      boost::random::discrete_distribution<Eigen::Index, double>>
+      rand_psis_idx(rng,
+                    boost::random::discrete_distribution<Eigen::Index, double>(
+                        boost::iterator_range<double*>(
+                            weight_vals.data(),
+                            weight_vals.data() + weight_vals.size())));
+  for (size_t i = 0; i <= num_multi_draws - 1; ++i) {
+    parameter_writer(samples.col(rand_psis_idx()));
+  }
+  const auto end_psis_time = std::chrono::steady_clock::now();
+  double psis_delta_time
+      = std::chrono::duration_cast<std::chrono::milliseconds>(end_psis_time
+                                                              - start_psis_time)
+            .count()
+        / 1000.0;
+  parameter_writer();
+  const auto time_header = std::string("Elapsed Time: ");
+  std::string optim_time_str = time_header
+                               + std::to_string(pathfinders_delta_time)
+                               + " seconds (Pathfinders)";
+  parameter_writer(optim_time_str);
```

Comment:
> Why is this going to the parameter-writer rather than the logger? 
We always append the end time to the csv as a comment
> Does this make it a comment in the output somehow?
Yes

Reply:
thanks.